Makefile.am

```diff
@@ -48,11 +48,17 @@ pkgconfigdir = @pkgconfigdir@
 # to build src/mpi/errhan/defmsg.h
 errnames_txt_files = 
 
-external_subdirs = @mplsrcdir@ @zmsrcdir@ @hwlocsrcdir@ @jsonsrcdir@ @yaksasrcdir@
-external_ldflags = @mpllibdir@ @zmlibdir@ @hwloclibdir@ @netloclibdir@ @yaksalibdir@
+external_subdirs = @mplsrcdir@ @zmsrcdir@ @hwlocsrcdir@ @jsonsrcdir@
+external_ldflags = @mpllibdir@ @zmlibdir@ @hwloclibdir@ @netloclibdir@
 external_libs = @WRAPPER_LIBS@
 mpi_convenience_libs =
-pmpi_convenience_libs = @mpllib@ @zmlib@ @hwloclib@ @jsonlib@ @yaksalib@
+pmpi_convenience_libs = @mpllib@ @zmlib@ @hwloclib@ @jsonlib@
+
+if BUILD_YAKSA_ENGINE
+external_subdirs += @yaksasrcdir@
+external_ldflags += @yaksalibdir@
+pmpi_convenience_libs += @yaksalib@
+endif
```

Comment:
`configure.ac` can simply set `@yaksasrcdir@` etc. to empty, right?

Reply:
Yes, we could do something like that. I think part of the issue is the yaksa configuration is split over 2 places. The main `configure.ac` and a `src/mpi/datatype/typerep/src/subconfigure.m4`. We should probably consolidate all the related so it is easier to grok. Will revise this PR.

Reply:
I see. I was thinking of merely adding a `AM_COND_IF` in `src/mpi/datatype/typerep/src/subconfigure.m4` and reset `yaksasrcdir` to empty there. I guess moving the `YAKSA` config from main `configure.ac` to `subconfigure` is another option -- probably better.

Reply:
Hmm, I have definitely wanted to use something like `AM_COND_IF` before, but I'm not sure I knew it existed! 😄

Reply:
Updated with 2 commits. 1 to move yaksa options to subconfigure.m4, and 2 to use `AM_COND_IF` to avoid setting up yaksa when not needed.